carladam/petrinet/arc.py

```diff
@@ -52,6 +49,17 @@ def default_arc_weight() -> ColorSet:
     return pmap({Abstract: 1})
 
 
+def weights_are_satisfied(arc: CompletedArcPT, tokens: AbstractSet[Token]) -> bool:
+    colors: ColorSet = Counter(token.color for token in tokens)
```

Comment:
Why is a ColorSet not a set?

This would make everything trivial for this function like:

ColorSet.from_tokens could instantiate the counter and build the set.

Arc.weight could be a ColorSet as well.

I think part of CarlAdams complexity is the lack of concrete concepts.

This is the thing I don't like about typing in Python… we describe structure instead of implementing concepts.